backend/globaleaks/handlers/accreditator/__init__.py

```diff
@@ -0,0 +1,158 @@
+            body_req,
+            requests.AccreditationInstructorRequest)
+        request['client_ip_address'] = self.request.client_ip
+        request['client_user_agent'] = self.request.client_ua
+        try:
+            request['organization_institutional_site'] = json.loads(body_req).get('organization_institutional_site')
+        except Exception as e:
+            logging.debug(e)
+            request['organization_institutional_site'] = None
+        return accreditation(request, is_instructor = True)
+
+class ToggleStatusActiveHandler(BaseHandler):
+    check_roles = 'any'
+    root_tenant_only = True
+    invalidate_cache = True
+
+    def put(self, accreditation_id: str):
+        return toggle_status_activate(accreditation_id, is_toggle=True)
+
+class AccreditationConfirmHandler(BaseHandler):
+    """
+    This manager is responsible for confirm accreditation requests
+    """
+    check_roles = 'any'
+    invalidate_cache = True
+    root_tenant_only = True
+
+    def post(self, accreditation_id: str):
+        request = self.validate_request(
+            self.request.content.read(),
+            requests.confirmAccreditation)
+        return activate_tenant(accreditation_id, request)
+
+class AccreditationApprovedHandler(BaseHandler):
+    """
+    This manager is responsible for confirm accreditation requests
+    """
+    check_roles = 'accreditor'
+    root_tenant_only = True
+
+    def post(self, accreditation_id: str):
+        return toggle_status_activate(accreditation_id)
+
+class AccreditationHandler(BaseHandler):
+    """
+    This manager is responsible for receiving accreditation requests and forwarding them to the accreditation manager
+    """
+    check_roles = 'any'
+    root_tenant_only = True
+    invalidate_cache = True
+
+    def get(self, accreditation_id: str):
+        return get_accreditation_by_id(accreditation_id)
+
+    def put(self, accreditation_id: str):
+        payload = self.request.content.read().decode('utf-8')
+        data = json.loads(payload) if payload else {}
+        return update_accreditation_by_id(accreditation_id, data)
+
+    def delete(self, accreditation_id: str):
+        request = self.validate_request(
+            self.request.content.read().decode('utf-8'),
+            requests.deleteAccreditation)
+        return persistent_drop(
+            accreditation_id,
+            request
+        )
+
+class GetAllAccreditationHandler(BaseHandler):
+    """
+    This manager is responsible for receiving accreditation requests and forwarding them to the accreditation manager
+    """
+    check_roles = 'accreditor'
+    root_tenant_only = True
+    invalidate_cache = True
+
+    def get(self):
+        return get_all_accreditation()
+
+class SubmitAccreditationHandler(BaseHandler):
+    """
+    This manager is responsible for receiving accreditation requests and forwarding them to the accreditation manager
+    """
+    check_roles = 'any'
```

Comment:
SECURITY ISSUE: any authenticated user can access this handler regardless of their permissions and not only accreditors